### PR TITLE
Apply methods last - Homework complete

### PR DESCRIPTION
**Pull request by glawson6**
{
"comfortability": 5,
"completeness": 5
}

### CODE REVIEW

File: talkativeRobotRGB.rb [hunk 1]
```diff
@@ -1,8 +1,29 @@
-puts("How old are you?")
-age = gets().chomp().to_i()
+def askAge
```
Comment (rgpass) [inline talkativeRobotRGB.rb:1]: Ruby uses snake_case 100% of the time.

File: talkativeRobotRGB.rb [hunk 2]
```diff
@@ -1,8 +1,29 @@
...
+  puts("How old are you?")
+  gets().chomp().to_i()
```
Comment (rgpass) [inline talkativeRobotRGB.rb:3]: Drop every `()`

File: talkativeRobotRGB.rb [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
+def askGender
+  puts("Are you male or female (M/F)?")
+  gets().chomp()
+end
 
-puts("Are you male or female (M/F)?")
-genderInput = gets().chomp()
+def lastMessage(ageDifference,gender)
+
+  if (ageDifference > 60 && gender == "M")
```
Comment (rgpass) [inline talkativeRobotRGB.rb:13]: Don't need the `()` around any of the `if` or `elsif` statements. It adds extra clutter.

File: talkative_task_robot.rb [hunk 4]
```diff
@@ -0,0 +1,83 @@
+#gerry.pass
```
Comment (rgpass) [inline talkative_task_robot.rb:1]: It should only be in one file. Each lab builds on the previous one.

File: talkative_task_robot.rb [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+#In ~/ruby_scripts, create a file called hashes.rb
+#Go through the entire Ruby doc on hashes and practice each method
+#Ignore any method that takes a block
+#Ex of one to ignore: hash.each { |k, v| puts “#{k} : #{v}” }
+#We’re covering these next lesson
+walk_the_dog = "walk the dog"
+to_dos = ["dishes", walk_the_dog, "get groceries", "create some code", "get some sleep"]
+
+first_task = to_dos.first
+puts(" First thing you need to do #{first_task}")
```
Comment (rgpass) [inline talkative_task_robot.rb:44]: Could combine these two lines and have it read nicer.

File: talkative_task_robot.rb [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+last_task = to_dos.last
+puts(" Last thing you need to do #{last_task}")
+wthedog_index = to_dos.index(walk_the_dog)
```
Comment (rgpass) [inline talkative_task_robot.rb:47]: Write out variable names. It makes it clearer and easier to maintain.

File: talkative_task_robot.rb [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+#Ignore any method that takes a block
+#Ex of one to ignore: hash.each { |k, v| puts “#{k} : #{v}” }
+#We’re covering these next lesson
+walk_the_dog = "walk the dog"
+to_dos = ["dishes", walk_the_dog, "get groceries", "create some code", "get some sleep"]
+
+first_task = to_dos.first
+puts(" First thing you need to do #{first_task}")
+last_task = to_dos.last
+puts(" Last thing you need to do #{last_task}")
```
Comment (rgpass) [inline talkative_task_robot.rb:46]: Curious: why the space after the start of the quote?

Also, don't need the `()` for puts.

File: talkative_task_robot.rb [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+wthedog_index = to_dos.index(walk_the_dog)
+puts(" What index is “walk dog”? => #{wthedog_index}")
+puts(" You did the last thing on the last #{to_dos.pop}")
+puts(" You just did the  #{to_dos.shift}")
+last_task = "last task"
+to_dos << last_task
+puts(" You just remembered you want to do something when all the other stuff is done #{last_task}")
+user = {}
+q1 = {question1: " First thing you need to do #{first_task}"}
```
Comment (rgpass) [inline talkative_task_robot.rb:55]: Don't need to have the key numbered (`question1`). If you don't have it numbered, you can easily enumerate through.

File: talkative_task_robot.rb [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+q2 = {question2: " Last thing you need to do #{last_task}"}
+q3 = {question3: " What index is “walk dog”? => #{wthedog_index}"}
+q4 = {question4: " You did the last thing on the last #{to_dos.pop}"}
+q5 = {question5: " You just did the #{to_dos.shift}"}
+getting_choice_arr = [true, false]
```
Comment (rgpass) [inline talkative_task_robot.rb:60]: Don't say the data type in the variable name.

Not sure what this variable is doing.

File: talkative_task_robot.rb [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+groceries = ["eggs", "flour", "milk", "sugar", "juice"]
+random_item = groceries.sample
+
+user.merge!(q1)
+user.merge!(q2)
+user.merge!(q3)
+user.merge!(q4)
+user.merge!(q5)
+user[:groceries] = groceries
```
Comment (rgpass) [inline talkative_task_robot.rb:70]: Nice.

File: talkative_task_robot.rb [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+getting_choice_arr = [true, false]
+
+groceries = ["eggs", "flour", "milk", "sugar", "juice"]
+random_item = groceries.sample
+
+user.merge!(q1)
+user.merge!(q2)
+user.merge!(q3)
+user.merge!(q4)
+user.merge!(q5)
```
Comment (rgpass) [inline talkative_task_robot.rb:69]: Would be much cleaner to have an array of strings for the questions.

**Comment by rgpass** [comment]
Requirements have been met, however putting some work into the style and organization will drastically help you later on.

File: talkative_task_robot.rb (hunk 8 above)
Comment (glawson6) [inline talkative_task_robot.rb:55]: if an array is easily iterated, then the keys of a hash should be easily iterated. 

```
 On Thursday, January 15, 2015 8:24 AM, Gerry Pass <notifications@github.com> wrote:
```

 In talkative_task_robot.rb:> +walk_the_dog = "walk the dog"

File: talkativeRobotRGB.rb (hunk 3 above)
Comment (glawson6) [inline talkativeRobotRGB.rb:13]: Although the () is optional, it is clearer to me what the intent is. I am calling the puts method. 

```
 On Thursday, January 15, 2015 8:20 AM, Gerry Pass <notifications@github.com> wrote:
```

 In talkativeRobotRGB.rb:>

**Comment by glawson6** [comment]
Style is a matter of opinion. Once we start getting into classes, a clear separation of concerns should start to emerge and hence better organization of code. 

```
 On Thursday, January 15, 2015 8:27 AM, Gerry Pass <notifications@github.com> wrote:
```

 Requirements have been met, however putting some work into the style and organization will drastically help you later on.—

**Pull request merged**